btree/btree.c

```diff
@@ -3529,21 +3566,417 @@ int64_t btree_iter_tick(struct m0_sm_op *smop)
 	}
 }
 
-int64_t btree_del_tick(struct m0_sm_op *smop)
+/* Delete Operation */
+
+/**
+ * This function will get called if there is an underflow at current node after
+ * deletion of the record. Currently, underflow condition is defined based on
+ * record count. If record count is 0, there will be underflow. To resolve
+ * underflow,
+ * 1) delete the node from parent.
+ * 2) check if there is an underflow at parent due to deletion of it's child.
+ * 3) if there is an underflow,
+ *        if, we have reached root, handle underflow at root.
+ *        else, repeat steps from step 1.
+ *    else, return next phase which needs to be executed.
+ *
+ * @param bop it will provide all required information about btree operation.
+ * @return int64_t it return state which needs to get executed next.
+ */
+static int64_t m0_btree_del_resolve_underflow(struct m0_btree_op *bop)
+{
+	struct td             *tree       = bop->bo_arbor->t_desc;
+	struct m0_btree_oimpl *oi         = bop->bo_i;
+	struct level          *lev;
+	struct slot            node_slot;
+	bool                   flag = false;
+	int                    curr_root_level;
+	struct slot            root_slot;
+	struct nd             *root_child;
+	int                    used_count = oi->i_used;
```

Comment:
Alignment.
Maybe keep the variables having assignment together for the sake of alignment.

Reply:
okay